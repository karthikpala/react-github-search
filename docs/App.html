<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: App</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: App</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>App<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">Component to render react app</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="App"><span class="type-signature"></span>new App<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Create an App component
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    
    <dt class="tag-author">Author:</dt>
    <dd class="tag-author">
        <ul>
            <li>[Karthik Pala]</li>
        </ul>
    </dd>
    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="main_App.jsx.html">main/App.jsx</a>, <a href="main_App.jsx.html#line15">line 15</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    
        <h3 class="subsection-title">Extends</h3>

        


    <ul>
        <li>Component</li>
    </ul>


    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    React render Method
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="main_App.jsx.html">main/App.jsx</a>, <a href="main_App.jsx.html#line22">line 22</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="APIServices.html">APIServices</a></li><li><a href="App.html">App</a></li><li><a href="RepoDetails.html">RepoDetails</a></li><li><a href="RepoList.html">RepoList</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Wed Jun 02 2021 21:32:02 GMT+0530 (India Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>