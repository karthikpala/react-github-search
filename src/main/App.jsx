import React, { Component } from 'react';
import RepoList from '../components/RepoList/RepoList';
import './app.scss';
import { CONFIG } from '../utils/constants';

/**
 * Component to render react app
 * @author [Karthik Pala]
 * @extends Component
 */
class App extends Component {
  /**
   * Create an App component
   */
  constructor(props) {
    super(props);
  }

  /**
   * React render Method
   */
  render() {
    return (
      <>
        <header className="flex flex--center app-header">
          <h2 className="app-header__title">
            <i className="fa fa-user-circle-o"></i> {CONFIG.title}
          </h2>
        </header>
        <main className="app-main">
          <div className="app-main__content">
            <RepoList />
          </div>
        </main>
      </>
    );
  }
}

export default App;
