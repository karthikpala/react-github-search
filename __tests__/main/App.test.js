import React from 'react';
import App from '../../src/main/App';
import repoResult from '../../__mocks__/repoResult.json';
import { create } from 'react-test-renderer';
import fetch from 'jest-fetch-mock';

/*
 * If nodes are rmeoved (or) some changes are made in the existing component, This test case will fail
 * We either have to update the snapshot by running node run undate-snapshots (or) fix the issue in component
 * Childer component can be either mocked (or) added with valid API calls mocked
 * I choose the latter one
 */
describe('App', () => {
  beforeEach(() => {
    fetch.resetMocks();
    window.IntersectionObserver = () => observeMock;
  });

  it('should render App component correctly', () => {
    fetch.mockResponseOnce(JSON.stringify(repoResult));
    const component = create(<App />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

export const observeMock = {
  observe: () => null
};
