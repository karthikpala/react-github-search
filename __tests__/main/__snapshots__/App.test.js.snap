// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component correctly 1`] = `
Array [
  <header
    className="flex flex--center app-header"
  >
    <h2
      className="app-header__title"
    >
      <i
        className="fa fa-user-circle-o"
      />
       
      Github Repo Search
    </h2>
  </header>,
  <main
    className="app-main"
  >
    <div
      className="app-main__content"
    >
      <div
        className="options"
      >
        <div
          className="options__left flex flex--left"
        />
        <div
          className="flex flex--center"
        >
          <div
            className="wrapper"
          >
            <input
              onChange={[Function]}
              placeholder="Search repositories"
              type="search"
            />
          </div>
        </div>
        <div
          className="options__right flex"
        />
      </div>
      <div
        className="flex flex--center empty"
      >
        <i
          className="empty__icon fa fa-search"
        />
        <p>
          Start searching by entering repo name 
        </p>
      </div>
      <div
        className="flex flex--center page-loading"
      >
        <span />
      </div>
    </div>
  </main>,
]
`;
